### DISCUSSION TITLE
Combining datasets

### DISCUSSION THREAD

**Discussion by leCoq42** [unrealities/MTGA_Draft_17Lands#57] [Ideas]
Hi @unrealities  @fiyir,

I want to suggest to add the possibility for users to combine datasets into a new dataset as a feature. This way users can download data for the top and middle tier 17lands players for example and then combine these into a new dataset containing the mid+top tier data. This gives the advantage of filtering out (bad) data from bad players. But still keep the advantage of having enough data to be useful in decision making.

I think that letting users download datasets separately is preferable over downloading everything at once, because otherwise downloading might fail due to the server rate limiting them. 

Unfortunately I currently don't have much time to make this feature myself. Also because om not familiar with programming gui's. Love to hear your opinions on this feature.

**Comment by unrealities** [comment]
I think I understand what you're saying, but I don't know how easy it would be to implement.

The user would still need to download each tier individually, but then there would be another button to create a custom file where they could select multiple files to combine.

What I'm not really sure about is whether you would need to recalculate the metrics with the new file, as all the percentages would change. I haven't touched on downloading files myself; I figured all the data is already in the files when you download them, and we're not doing any math on our end.

**Reply by leCoq42** [reply]
I've started working on it. For the first version the user has two download the sets separately first indeed and then combine them via a button in the ui. 

The math would not be hard I think, you just have to weight the metrics based on the count for each metric.
For example:
If you have for top players #gih 100, gihwr 60%, for middle players #gih 100, gihwr 55%.
Then the when combining the set it should perform the following calculation: (100x60%+100x55%)/200=57.5%

I might need some help to get this implemented in the ui at some point, but I'll try to work out the combining part out for now.
